apps/api/v2/src/modules/slots/controllers/slots.controller.ts

```diff
@@ -129,6 +158,44 @@ export class SlotsController {
       },
     });
 
+    if (query.formatAsStartAndEndTime) {
+      let duration = 0;
+      if (query.duration) {
+        duration = query.duration;
+      } else if (query.eventTypeId) {
+        const eventType = await this.slotsService.getEventTypeWithDuration(query.eventTypeId);
+
+        if (!eventType) {
+          throw new Error("Event type not found");
+        }
+
+        duration = eventType.length;
+      } else {
+        throw new Error("duration or eventTypeId is required");
+      }
+
+      const transformedSlots = Object.entries(availableSlots.slots).reduce<
+        Record<string, { startTime: string; endTime: string }[]>
+      >((acc, [date, slots]) => {
+        acc[date] = (slots as { time: string }[]).map((slot) => {
+          const startTime = new Date(slot.time);
+          const endTime = new Date(startTime.getTime() + duration * 60000); 
+          return {
+            startTime: startTime.toISOString(),
+            endTime: endTime.toISOString(),
+          };
+        });
+        return acc;
+      }, {});
```

Comment:
place this whole function in slots.service, name as formatSlots for example, have it return the formatted slots